proposals/dictionary-expressions.md

```diff
@@ -308,6 +308,52 @@ If the target type is a *struct* or *class type* that implements `System.Collect
       * If the enumerator implements `IDisposable`, then `Dispose` will be called after enumeration, regardless of exceptions.
       * ...
 
+If the target type is an *array*, a *span*, a type with a *[create method](#create-methods)*, or an *interface*, the construction of the collection instance is as follows:
+
+* The elements are evaluated in order. Some or all elements may be evaluated *during* the steps below rather than before.
+
+* The compiler *may* determine the *known length* of the collection expression by invoking [*countable*](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-8.0/ranges.md#adding-index-and-range-support-to-existing-library-types) properties &mdash; or equivalent properties from well-known interfaces or types &mdash; on each *spread element expression*.
+
+* An *initialization instance* is created as follows:
+  * If the target type is an *array* and the collection expression has a *known length*, an array is allocated with the expected length.
+  * If the target type is a *span* or a type with a *create method*, and the collection has a *known length*, a span with the expected length is created referring to contiguous storage.
+  * Otherwise intermediate storage is allocated.
+
+* **If the *iteration type* is a type `KeyValuePair<K, V>`, then:**
+  * **For each element in order:**
+    * **If the element is a *key value pair element* `Kᵢ:Vᵢ` then:**
+      * **First `Kᵢ` is evaluated, then `Vᵢ` is evaluated.**
+      * **The indexer is invoked on the collection instance with the converted values of `Kᵢ` and `Vᵢ`.**
```

Comment:
what if there is no indexer?  I think i'm missing where we stated we know the dest type has that here.

Reply:
This section applies:
> If the target type is an *array*, a *span*, a type with a *create method*, or an *interface*

In all of those cases, we have an indexer. (In each of those cases, we're either constructing either an array or intermediate storage with a type that has an indexer.)

Reply:
k